test/integration_test.ts

```diff
@@ -5,6 +5,15 @@ import * as path from 'path';
 import * as main from '../lib/main';
 
 describe('integration test: public api', () => {
+  let _warn = console.warn;
```

Comment:
this is kind of dangerous. if you are going to mess with globals, then you should capture the value in beforeEach and overwrite it, then in after each restore it.

if you capture the value in describe and some other file will be doing the same kind of patching, you will collide.